functions/go/set-annotation/Dockerfile

```diff
@@ -0,0 +1,8 @@
+FROM golang:1.13-stretch
```

Comment:
We have started to use `golang:1.15-alpine3.12` and `node:14.15-alpine3.12` as build image and `alpine:3.12` as the final base image in the whole repo. e.g. 
https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/set-namespace/Dockerfile

Please change this file and also https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/set-label/Dockerfile.